polkadot/api/src/full.rs

```diff
@@ -20,45 +20,55 @@ use client::backend::{Backend, LocalBackend};
 use client::block_builder::BlockBuilder as ClientBlockBuilder;
 use client::{Client, LocalCallExecutor};
 use polkadot_executor::Executor as LocalDispatch;
-use substrate_executor::NativeExecutor;
 use state_machine;
+use substrate_executor::NativeExecutor;
 
+use primitives::parachain::{CandidateReceipt, DutyRoster, Id as ParaId};
+use primitives::{
+	AccountId, Block, BlockId, Hash, Header, Index, SessionKey, Timestamp, UncheckedExtrinsic,
+};
 use runtime::Address;
 use runtime_primitives::traits::AuxLookup;
-use primitives::{AccountId, Block, Header, BlockId, Hash, Index, SessionKey, Timestamp, UncheckedExtrinsic};
-use primitives::parachain::{CandidateReceipt, DutyRoster, Id as ParaId};
 
-use {BlockBuilder, PolkadotApi, LocalPolkadotApi, ErrorKind, Error, Result};
+use {BlockBuilder, Error, ErrorKind, LocalPolkadotApi, PolkadotApi, Result};
 
 // set up the necessary scaffolding to execute a set of calls to the runtime.
 // this creates a new block on top of the given ID and initialises it.
 macro_rules! with_runtime {
-	($client: ident, $at: expr, $exec: expr) => {{
+	($client:ident, $at:expr, $exec:expr) => {{
 		let parent = $at;
 		let header = Header {
-			parent_hash: $client.block_hash_from_id(&parent)?
+			parent_hash: $client
+				.block_hash_from_id(&parent)?
 				.ok_or_else(|| ErrorKind::UnknownBlock(format!("{:?}", parent)))?,
-			number: $client.block_number_from_id(&parent)?
+			number: $client
+				.block_number_from_id(&parent)?
 				.ok_or_else(|| ErrorKind::UnknownBlock(format!("{:?}", parent)))? + 1,
 			state_root: Default::default(),
 			extrinsics_root: Default::default(),
 			digest: Default::default(),
-		};
-
-		$client.state_at(&parent).map_err(Error::from).and_then(|state| {
-			let mut changes = Default::default();
-			let mut ext = state_machine::Ext::new(&mut changes, &state);
-
-			::substrate_executor::with_native_environment(&mut ext, || {
-				::runtime::Executive::initialise_block(&header);
-				($exec)()
-			}).map_err(Into::into)
-		})
-	}}
+			};
+
+		$client
+			.state_at(&parent)
+			.map_err(Error::from)
+			.and_then(|state| {
+				let mut changes = Default::default();
+				let mut ext = state_machine::Ext::new(&mut changes, &state);
+
+				::substrate_executor::with_native_environment(&mut ext, || {
+					::runtime::Executive::initialise_block(&header);
+					($exec)()
+				}).map_err(Into::into)
+			})
+		}};
 }
 
-impl<B: LocalBackend<Block>> BlockBuilder for ClientBlockBuilder<B, LocalCallExecutor<B, NativeExecutor<LocalDispatch>>, Block>
-	where ::client::error::Error: From<<<B as Backend<Block>>::State as state_machine::backend::Backend>::Error>
+impl<B: LocalBackend<Block>> BlockBuilder
+	for ClientBlockBuilder<B, LocalCallExecutor<B, NativeExecutor<LocalDispatch>>, Block>
+where
+	::client::error::Error:
```

Comment:
this is an example of how the `where` indentation works in rustfmt within the context and I think this is a decent way of addressing that...

Reply:
I think it's worse than the original.

Reply:
But is your complaint about `where` being at the beginning of the line or because it is breaking the line prior? Considering that this line, too, is a whopping 128chars, it is violation even if we adhered to the current style guide and the current style guide doesn't really say where `for` needs to break here or how this case should be considered.